[Max Ng] iP

maxng17 opened the pull request:
### DukePro
> “Your mind is for having ideas, not holding them.” – David Allen [(source)](https://dansilvestre.com/productivity-quotes)

DukePro frees your mind of having to remember things you need to do. It's,
- text-based
- easy to learn
- ~~FAST~~ *SUPER FAST* to learn

All you need to do is,
1. down it from [here](https://nus-cs2103-ay2223s1.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😄 

And it is **FREE**!

Features:

- [X] Task list 1
- [ ] Task list 2 \(Coming soon)
- [ ] Task list 3 \(Coming soon)
:+1::+1::shipit::+1::+1:

```
public static void main(String args[]) {
    Application.launch(MainApp.class, args);
}
```
Line
- nest line
   - nest nest line

Text that is not a quote

> Text that is a quote

[Press here for fun](https://nus-cs2103-ay2223s1.github.io/website/schedule/week4/project.html)

maxng17 committed:
Replaced Duke.java with Blink.java
Replace instances of Duke with Blink in other files

maxng17 committed:
added Greet, Echo, Exit

maxng17 committed:
added Add, List

maxng17 committed:
added Mark as Done
* Added Task file
* Added mark and unmark commands
* Check for edge cases but no exception handling yet

maxng17 committed:
Add different types of tasks
* Add Deadlines
* Add Events
* Add ToDos

maxng17 committed:
Add ToDo, Event, Deadline
* Handle new commands (e.g. event)
* Handle non-valid commands

maxng17 committed:
Add A-TextUiTesting
* Added auto testing file for regression testing

maxng17 committed:
Add Handle Error
* Add BlinkException
* Methods in Blink and Task throw exceptions

maxng17 committed:
Add Delete
* Changed store array to arraylist
* Add delete functionality for tasks

maxng17 committed:
Add Use Enums
* Group commands into enum class
* Change some methods to private

maxng17 committed:
Updating of Test files

Test files input.txt and EXPECTED.TXT are outdated
and do not test for commands and exception added

Let's update the test by testing for delete command and
every type of exception that can be thrown currently.

maxng17 committed:
Add Save feature to save list into file

maxng17 committed:
Add Dates and Times for tasks

maxng17 committed:
Merge branch 'branch-Level-7'

maxng17 committed:
Merge branch 'branch-Level-8'

# Conflicts:
#	src/main/java/Blink.java
#	src/main/java/Task.java
#	src/main/java/ToDos.java

maxng17 committed:
Add more OOP by adding more classes

maxng17 committed:
Organize into Packages
* blink package for all files
* command package in Blink package
* task package in Blink package

maxng17 committed:
Revert "Organize into Packages"

This reverts commit 7d7d7d42b45fac84a4a6fd5d511c4e9ad8b6059e.

maxng17 committed:
Organize files into packages again

maxng17 committed:
Add Gradle

maxng17 committed:
Update values in build.gradle file

maxng17 committed:
Add JUnit test

maxng17 committed:
Add JavaDocs for all files

maxng17 committed:
Add Coding standard to all files

maxng17 committed:
Add Find command

maxng17 committed:
Merge branch 'branch-A-JavaDoc'

maxng17 committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/blink/TaskList.java
#	src/main/java/blink/command/Command.java
#	src/main/java/blink/command/ListCommand.java
#	src/main/java/blink/task/Deadlines.java
#	src/main/java/blink/task/Events.java
#	src/main/java/blink/task/ToDos.java

maxng17 committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/blink/TaskList.java
#	src/main/java/blink/Ui.java
#	src/main/java/blink/task/Deadlines.java

maxng17 committed:
Improve find command logic

maxng17 committed:
Add Checkstyle for gradle

kianseong commented on src/main/java/blink/TaskList.java:
```diff
@@ -0,0 +1,207 @@
+     *
+     * @return String containing number of remaining task in TaskList
+     */
+    public String deleted() {
```
Could rename this method? (to sound more like a verb and clearer about what the method does)

danzzzerl replied:
Agree with this point, was a little confusing when i saw it at first.

kianseong commented on src/main/java/blink/TaskList.java:
```diff
@@ -0,0 +1,207 @@
+     * @return ArrayList containing all tasks with same date
+     * as the one specified
+     */
+    public ArrayList<Task> filter(LocalDate date) {
+        ArrayList<Task> sameDates = new ArrayList<>();
+        for (int x = 0; x < this.length(); x++) {
+            Task temp = this.get(x);
+            if (temp.checkDate(date)) {
+                sameDates.add(temp);
+            }
+        }
+        return sameDates;
+    }
```
👍

kianseong commented on src/main/java/blink/task/Events.java:
```diff
@@ -0,0 +1,66 @@
+ * Events task that contains a description, boolean to indicate if
+ * it has been marked and date.
+ */
+public class Events extends Task{
```
Should the class name be singular (i.e Event)?
Should there a space before the opening brace as well?

kianseong commented on src/main/java/blink/Ui.java:
```diff
@@ -0,0 +1,153 @@
+     * @param tasks TaskList of all the current Tasks
+     * @param num Number position of Task to mark
+     */
+    public void mark(TaskList tasks, int num) {
```
Could this method be renamed? (i.e. showMarkedTask)
Likewise for the ones below.

danzzzerl replied:
Yup, TaskList has functions with similar names so it might be hard for you at especially at a later stage.

kianseong reviewed (commented):
LGTM! Other than some naming issues, coding standard 👍

danzzzerl commented on src/main/java/blink/Blink.java:
```diff
@@ -0,0 +1,67 @@
+     * are suitable commands and act upon them if they are.
+     */
+    public void run() {
+        ui.showWelcome();
```
I think a better method name would be printWelcomeMessage. But showWelcome could do as well.

danzzzerl commented on src/main/java/blink/TaskList.java:
```diff
@@ -0,0 +1,207 @@
+     * @param num Number position of task to mark
+     */
+    public void mark(int num) {
+        if (num <= 0 || num > this.length()) {
```
Good job on this guard clause, i think most people miss this part out and put the marking function first! Similar for the unMark function.

danzzzerl commented on src/main/java/blink/command/Command.java:
```diff
@@ -0,0 +1,27 @@
+/**
+ *  Command object that is created when a recognized command is input.
+ */
+abstract public class Command {
```
Quite a cool idea in making the Command class an abstract class! I think not many people do it 👍

danzzzerl reviewed (commented):
Neat code and easy to read! Other than some naming of methods, looks good! Great job brother.

maxng17 committed:
Add default GUI

maxng17 committed:
Merge branch 'branch-Level-10'

# Conflicts:
#	src/main/java/blink/Blink.java

maxng17 committed:
Clean up unused code

maxng17 committed:
Fix coding standard
The code does not follow the coding standard stated
in the CS2103T website.

By ensuring that the code follows a coding standard, it is easier
to read the code and also spot errors.

Lets use the checkstyle command in gradle to find coding standard
violations and rectify them.

maxng17 committed:
Use Assertions in TaskList and MainWindow
As TaskList methods are used in some commands, there is a
need to ensure that what it returns is correct. Since the parameters
passed affects what is returned, checking the parameters are
important too. For MainWindow, assertion is will be used to ensure that
for every command, an output is given.

Using Java Assertions allows us to check for runtime errors created
by the programmer. We will then be able to find any error in the code
or future errors that may be caused due to future changes.

Lets add assertion in different methods that are commonly called.

maxng17 committed:
Improve code quality
There are some unnecessary code and complicated expressions
used.

To improve readability it is important to improve the code
quality. Also complicated expressions are broken down into
different variables so that errors can be spotted easily.

Lets remove unused code and also identify complicated
expression that uses a mix of function calls and hardcoded values.
The complicated expression will be broken into different variables.

maxng17 committed:
Merge pull request #1 from maxng17/branch-A-Assertions

Use Assertions in TaskList and MainWindow

maxng17 committed:
Merge branch 'master' into branch-A-CodeQuality

# Conflicts:
#	src/main/java/blink/TaskList.java
#	src/main/java/blink/Ui.java

maxng17 committed:
Merge pull request #2 from maxng17/branch-A-CodeQuality

Improve code quality

maxng17 committed:
Set up CICD

maxng17 committed:
Add Tagging for task

maxng17 committed:
Merge branch 'A-Tagging'

maxng17 committed:
Improve personality of chatbot

maxng17 committed:
Add user guide

maxng17 committed:
Fix coding standard

maxng17 committed:
Add Ui.png and fix readME

maxng17 committed:
Add Title to Main window

maxng17 committed:
Improve UI.png image

maxng17 committed:
Change text font for Blink
